packages/core-transaction-pool/lib/interface.js

```diff
@@ -199,20 +199,30 @@ module.exports = class TransactionPoolInterface {
    */
   acceptChainedBlock(block) {
     for (const transaction of block.transactions) {
-      const senderWallet = this.walletManager.exists(
-        transaction.senderPublicKey,
-      )
-        ? this.walletManager.findByPublicKey(transaction.senderPublicKey)
+      const senderPublicKey = transaction.senderPublicKey
+
+      const senderWallet = this.walletManager.exists(senderPublicKey)
+        ? this.walletManager.findByPublicKey(senderPublicKey)
         : false
 
-      // if wallet in pool we try to apply transaction
+      const recipientWallet = this.walletManager.exists(transaction.recipientId)
+        ? this.walletManager.findByAddress(transaction.recipientId)
+        : false
+
+      // If sender or recipient wallet in pool we try to apply transaction
       const exists = this.transactionExists(transaction.id)
-      if (senderWallet && !exists) {
+      if (!exists && (senderWallet || recipientWallet)) {
         try {
           this.walletManager.applyPoolTransaction(transaction)
         } catch (error) {
           logger.error(`AcceptChainedBlock in pool: ${error}`)
-          this.purgeByPublicKey(transaction.senderPublicKey)
+          // Purge sender
+          this.purgeByPublicKey(senderPublicKey)
+
+          // Purge recipient if not sender
+          if (senderPublicKey !== recipientWallet.publicKey) {
+            this.purgeByPublicKey(recipientWallet.publicKey)
+          }
```

Comment:
Why remove also recipient's transactions?

This way somebody may deprive somebody else from sending anything by constantly purging all his transactions via this code. For example, A periodically generates a transaction A->B such that it would end up here, doing line 224. We remove all B's transactions from the pool. The innocent B is never able to send anything to anybody because his transactions are constantly being removed.